### ISSUE TITLE
ZFS broken on latest linux (>5.10)

### ISSUE THREAD

**Issue by rien** [NixOS/nixpkgs#150517]
### Describe the bug

ZFS (both stable and unstable) are currently marked as broken on linux version 5.15 because it isn't mentioned as supported in the latest release.

Version 5.14 was marked as supported, but since https://github.com/NixOS/nixpkgs/pull/147446, this version is EOL and is removed, leaving the latest supported version on 5.10 (broken on my own system).

There has been a PR merged into zfs' master  (https://github.com/openzfs/zfs/pull/12824) claiming support for 5.15, but I don't know if it is wise to include this in `nixpkgs` yet.

I've made this issue to
- 1. track when this is resolved,
- 2. serve as easy-to-find issue for others with the same problem,
- 3. ask whether it is a good idea to update `zfsUnstable` to zfs' latest commit on master. 

For what it's worth: @Mic92 has a patch which enables ZFS for 5.15 and they claim that it is running just fine https://github.com/Mic92/nixpkgs/commit/b31e0be77366618eab2a3f7a0648ec1786504732.

### Steps To Reproduce
1. Use ZFS
2. Set `boot.kernelPackages` to `linuxPackages_latest` or `linuxPackages_5_15`.
3. Build the system
4. It fails with the message that `zfs` is broken.

### Relevant issues/PR's
- https://github.com/openzfs/zfs/issues/12786
- https://github.com/NixOS/nixpkgs/pull/146310

### Notify maintainers

@hmenke @jcumming @jonringer @wizeman @fpletz @globin 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.14.21, NixOS, 22.05 (Quokka)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.4`
 - channels(rien): `""`
 - nixpkgs: `/etc/nixpkgs`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
 - zfs
 - zfsUnstable
 - linuxPackages_5_15
# a list of nixos modules affected by the problem
module:
 - boot.zfs
```

**Comment by hmenke** [comment]
I've submitted a feature request on OpenZFS: https://github.com/openzfs/zfs/issues/12847

**Comment by Mic92** [comment]
Yep. It's working for over a month. You can cherry-pick this commit if you want:

https://github.com/Mic92/nixpkgs/commit/b31e0be77366618eab2a3f7a0648ec1786504732

archlinux zfs user likely also using these kernel versions for quite a while.

**Comment by dtzWill** [comment]
Anyone have an overlay snippet or something to accomplish this? Would rather not start carrying patches over nixpkgs in my own fork again, "yet", I saw part of one (overlay-based solution) on another thread for this issue, but wasn't sure how to complete it or if it's actually been used successfully... Anyway if anyone has something to help there that'd be swell, thanks! :)

**Comment by hmenke** [comment]
@dtzWill Just set
```nix
{
  nixpkgs.allowBroken = true;
}
```
or if you are scared of that for whatever reason use
```nix
{
  boot.kernelPackages = linuxPackages_latest.extend (final: prev: {
    zfs = prev.zfs.overrideAttrs (_: { meta.broken = false; });
  });
}
```

**Comment by TredwellGit** [comment]
https://github.com/NixOS/nixpkgs/pull/150923

**Comment by fpletz** [comment]
Fixed by #150923.
